developer/images/dependencies-update/tasks/update_dockerfiles_base_images.sh

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env bash
+
+# Copyright 2022 The Pipeline Service Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+usage() {
+  echo "
+Usage:
+    ${0##*/} [options]
+
+Update the sha of the base images in the Dockefiles.
+
+Optional arguments:
+    -d, --debug
+        Activate tracing/debug mode.
+    -h, --help
+        Display this message.
+
+Example:
+    ${0##*/}
+" >&2
+}
+
+parse_args() {
+  while [[ $# -gt 0 ]]; do
+    case $1 in
+    -d | --debug)
+      set -x
+      DEBUG="--debug"
+      export DEBUG
+      ;;
+    -h | --help)
+      usage
+      exit 0
+      ;;
+    *)
+      echo "Unknown argument: $1"
+      usage
+      exit 1
+      ;;
+    esac
+    shift
+  done
+}
+
+process_dockerfiles() {
+  mapfile -t DOCKERFILES < <(
+    find "$PROJECT_DIR" -type f -name Dockerfile |
+      sed "s:$PROJECT_DIR/::" |
+      grep --invert-match --extended-regexp "developer/exploration" |
+      sort
+  )
+  while read -r from_base_image; do
+    process_base_image_cmd
+  done < <(
+    grep --no-filename --regexp "^#@FROM " "${DOCKERFILES[@]}" |
+    sort -u
+  )
+}
+
+process_base_image_cmd() {
+  base_image_name=$(echo "$from_base_image" | sed 's:^.* ::')
+  echo -n "- $base_image_name"
+  get_base_image_sha
+  echo -n "@$base_image_sha : "
+  if grep --max-count=1 --quiet "FROM $base_image_name@$base_image_sha" "${DOCKERFILES[@]}"; then
+    echo "No update"
+  else
+    update_base_image_sha
+    echo "Updated to latest sha"
+  fi
+}
+
+get_base_image_sha() {
+  base_image_sha=$(skopeo inspect "docker://$base_image_name" | yq ".Digest")
```

Comment:
We need to add skopeo as one of dependent tools in [DEPENDENCIES.md](https://github.com/openshift-pipelines/pipeline-service/blob/main/DEPENDENCIES.md) and [dependencies.sh](https://github.com/openshift-pipelines/pipeline-service/blob/main/shared/config/dependencies.sh)

Reply:
Add to [DEPENDENCIES.md](https://github.com/openshift-pipelines/pipeline-service/blob/main/DEPENDENCIES.md) only.  Skopeo is currently not installable through [install.sh](https://github.com/openshift-pipelines/pipeline-service/blob/main/shared/hack/install.sh), so the version does not need to be fixed in [dependencies.sh](https://github.com/openshift-pipelines/pipeline-service/blob/main/shared/config/dependencies.sh).